File: Assets/Plugins/VoxelBusters/EssentialKit/Runtime/Core/BillingServices/BillingProductSubscriptionExpirationReason.cs
namespace VoxelBusters.EssentialKit
{
    /// <summary>
    /// Reason for billing product subscription expiration
    /// </summary>
    public enum BillingProductSubscriptionExpirationReason : int
    {
        /// <summary>
        /// No expiration reason available.
        /// </summary>
        None,

        /// <summary>
        /// Unknown expiration reason.
        /// </summary>
        Unknown,

        /// <summary>
        /// Auto-Renew disabled.
        /// </summary>
        AutoRenewDisabled,

        /// <summary>
        /// Expiration due to billing error.
        /// </summary>
        BillingError,

        /// <summary>
        /// Expiration due to user did not consent to price increase.
        /// </summary>
        DidNotConsentToPriceIncrease,

        /// <summary>
        /// Expiration due to product being unavailable.
        /// </summary>
        ProductUnavailable
    }


}

File: Assets/Plugins/VoxelBusters/EssentialKit/Plugins/iOS/BillingServices/Modules/BillingServices/Enums/BillingProductSubscriptionExpirationReason.swift
//
//  BillingProductSubscriptionExpirationReason.swift
//  BillingServices
//
//  Created by Ayyappa on 04/04/24.
//

import Foundation

@objc
public enum BillingProductSubscriptionExpirationReason: Int {
    case none
    case unknown
    case autoRenewDisabled
    case billingError
    case didNotConsentToPriceIncrease
    case productUnavailable
}
